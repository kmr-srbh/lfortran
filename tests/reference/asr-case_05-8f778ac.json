{
    "basename": "asr-case_05-8f778ac",
    "cmd": "lfortran --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/case_05.f90",
    "infile_hash": "914477f4a376baf0341e522d2109b6108238cd1d9dfecf969283ab79",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-case_05-8f778ac.stdout",
    "stdout_hash": "a0a297daa7fe82ffeed3807c193ef6239db0cbbd59eabec3d673b565",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}