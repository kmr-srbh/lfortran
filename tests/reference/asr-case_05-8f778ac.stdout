(TranslationUnit
    (SymbolTable
        1
        {
            case_05:
                (Program
                    (SymbolTable
                        2
                        {
                            grade:
                                (Variable
                                    2
                                    grade
                                    []
                                    Local
                                    (StringConstant
                                        "B"
                                        (Character 1 1 () PointerString)
                                    )
                                    (StringConstant
                                        "B"
                                        (Character 1 1 () PointerString)
                                    )
                                    Save
                                    (Character 1 1 () PointerString)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    case_05
                    []
                    [(Select
                        (Var 2 grade)
                        [(CaseStmt
                            [(StringConstant
                                "A"
                                (Character 1 1 () PointerString)
                            )]
                            [(Print
                                [(StringConstant
                                    "Excellent!"
                                    (Character 1 10 () PointerString)
                                )]
                                ()
                                ()
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(StringConstant
                                "B"
                                (Character 1 1 () PointerString)
                            )]
                            []
                            .false.
                        )
                        (CaseStmt
                            [(StringConstant
                                "C"
                                (Character 1 1 () PointerString)
                            )]
                            [(Print
                                [(StringConstant
                                    "Well done"
                                    (Character 1 9 () PointerString)
                                )]
                                ()
                                ()
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(StringConstant
                                "D"
                                (Character 1 1 () PointerString)
                            )]
                            [(Print
                                [(StringConstant
                                    "You passed"
                                    (Character 1 10 () PointerString)
                                )]
                                ()
                                ()
                            )]
                            .false.
                        )
                        (CaseStmt
                            [(StringConstant
                                "F"
                                (Character 1 1 () PointerString)
                            )]
                            [(Print
                                [(StringConstant
                                    "Better try again"
                                    (Character 1 16 () PointerString)
                                )]
                                ()
                                ()
                            )]
                            .false.
                        )]
                        [(Print
                            [(StringConstant
                                "Invalid grade"
                                (Character 1 13 () PointerString)
                            )]
                            ()
                            ()
                        )]
                        .false.
                    )
                    (Print
                        [(StringFormat
                            ()
                            [(StringConstant
                                "Your grade is "
                                (Character 1 14 () PointerString)
                            )
                            (Var 2 grade)]
                            FormatFortran
                            (Character -1 0 () PointerString)
                            ()
                        )]
                        ()
                        ()
                    )]
                )
        })
    []
)
